### PR TITLE
Add ability to hide hidden datasets

### PR DESCRIPTION
**Pull request by pchaussalet**
Ticket: #20453

### CODE REVIEW

File: nightwatch.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "test_settings": {
         "default": {
             "filter": "index.js",
-            "launch_url": "http://localhost:3000/#!/?host=extrasuck.ixsystems.com",
```
Comment (thibaultzanini) [inline nightwatch.json]: Is it really related to that PR?

**Review by thibaultzanini — COMMENTED** [review]
except a comment LGTM

File: core/model/user-interface-descriptors/volume-dataset-user-interface-descriptor.mjson [hunk 2]
```diff
@@ -4,6 +4,8 @@
         "properties": {
             "inspectorComponentModule": {"%": "ui/sections/storage/inspectors/volume-dataset.reel"},
             "creatorComponentModule": {"%": "ui/sections/storage/inspectors/volume-dataset.reel"},
+            "listControlsComponentModule": {"%": "ui/sections/storage/controls/volume-dataset-options.reel"},
+            "listControlsFilterExpression": "name.substring(volume.length)[0] != '.'",
```
Comment (thibaultzanini) [inline core/model/user-interface-descriptors/volume-dataset-user-interface-descriptor.mjson]: Needs to be removed

**Comment by pchaussalet** [comment]
Updated

**Pull request merged**